taurus/entity/dict_serializable.py

```diff
@@ -82,7 +82,40 @@ def build(d):
         return loads(dumps(d))
 
     def __repr__(self):
-        return str(self.as_dict())
+        object_dict = self.as_dict()
+        # as_dict() skips over keys in `skip`, but they should be in the representation.
+        skipped_keys = {x.lstrip('_') for x in vars(self) if x in self.skip}
+        for key in skipped_keys:
+            skipped_field = self.__getattribute__(key)
+            object_dict[key] = self._name_repr(skipped_field)
+        return object_dict.__str__()
```

Comment:
Similarly, you can use `str(object_dict)` -- in general any built-in function defined like `__foo__(self)` can be invoked with `foo(object_implementing___foo__)`

Reply:
Done